apps/cowswap-frontend/src/common/hooks/useGetSurplusFiatValue.ts

```diff
@@ -39,13 +40,16 @@ export function useGetSurplusData(order: Order | ParsedOrder | undefined): Outpu
 
   const showSurplus = shouldShowSurplus(surplusFiatValue, surplusAmount)
 
-  return {
-    surplusFiatValue,
-    showFiatValue,
-    surplusToken,
-    surplusAmount,
-    showSurplus,
-  }
+  return useSafeMemo(
+    () => ({
+      surplusFiatValue,
+      showFiatValue,
+      surplusToken,
+      surplusAmount,
+      showSurplus,
+    }),
+    [surplusFiatValue, showFiatValue, surplusToken, surplusAmount, showSurplus]
```

Comment:
How this is solving the issue?

I mean, this is just memorizing the result.
if it was true the theory that this is because `showSurplus` is true, then false, then true again (because of the USD), how is meoizing helping?

Reply:
🤷